packages/vite/src/node/optimizer/index.ts

```diff
@@ -597,84 +597,71 @@ export function runOptimizeDeps(
 
   const start = performance.now()
 
-  const preparedRun = prepareEsbuildOptimizerRun(
+  const preparedRun = prepareRollupOptimizerRun(
     resolvedConfig,
     depsInfo,
     ssr,
     processingCacheDir,
     optimizerContext,
   )
 
-  const runResult = preparedRun.then(({ context, idToExports }) => {
-    function disposeContext() {
-      return context?.dispose().catch((e) => {
-        config.logger.error('Failed to dispose esbuild context', { error: e })
-      })
-    }
-    if (!context || optimizerContext.cancelled) {
-      disposeContext()
+  const runResult = preparedRun.then(({ build, idToExports }) => {
+    if (!build || optimizerContext.cancelled) {
       return cancelledResult
     }
 
-    return context
-      .rebuild()
+    return build()
       .then((result) => {
-        const meta = result.metafile!
-
-        // the paths in `meta.outputs` are relative to `process.cwd()`
+        // TODO: Make sure the paths in `meta.outputs` are relative to `process.cwd()`
         const processingCacheDirOutputPath = path.relative(
           process.cwd(),
           processingCacheDir,
         )
 
-        for (const id in depsInfo) {
-          const output = esbuildOutputFromId(
-            meta.outputs,
-            id,
-            processingCacheDir,
-          )
-
-          const { exportsData, ...info } = depsInfo[id]
-          addOptimizedDepInfo(metadata, 'optimized', {
-            ...info,
-            // We only need to hash the output.imports in to check for stability, but adding the hash
-            // and file path gives us a unique hash that may be useful for other things in the future
-            fileHash: getHash(
-              metadata.hash +
-                depsInfo[id].file +
-                JSON.stringify(output.imports),
-            ),
-            browserHash: metadata.browserHash,
-            // After bundling we have more information and can warn the user about legacy packages
-            // that require manual configuration
-            needsInterop: needsInterop(
-              config,
-              ssr,
-              id,
-              idToExports[id],
-              output,
-            ),
-          })
-        }
-
-        for (const o of Object.keys(meta.outputs)) {
-          if (!o.match(jsMapExtensionRE)) {
-            const id = path
-              .relative(processingCacheDirOutputPath, o)
-              .replace(jsExtensionRE, '')
-            const file = getOptimizedDepPath(id, resolvedConfig, ssr)
-            if (
-              !findOptimizedDepInfoInRecord(
-                metadata.optimized,
-                (depInfo) => depInfo.file === file,
-              )
-            ) {
-              addOptimizedDepInfo(metadata, 'chunks', {
-                id,
-                file,
-                needsInterop: false,
+        for (const chunk of result.output) {
```

Comment:
Vite using esbuild `metafile.outputs` to collect optimized metadata, here using `RollupOutput` instead of it. Rolldown also need implement this, track here https://github.com/rolldown-rs/rolldown/issues/185